site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -146,69 +181,263 @@ def _transform_publication_date(self):
+                rdm_copyright = "© " + ", ".join(parts)
+
+                result_list.append(rdm_copyright)
+        return "<br />".join(result_list)
+
+    def _transform_dois(self):
+        """Mapping of record dois."""
+        DATACITE_PREFIX = current_app.config["DATACITE_PREFIX"]
+        dois = self.inspire_metadata.get("dois", [])
+
+        if len(dois) > 1:
+            self.metadata_errors.append(
+                f"More than 1 DOI was found in the INSPIRE record #{self.inspire_metadata.get('control_number')}."
+            )
+            return None
+        elif len(dois) == 0:
+            return None
+        else:
+            doi = dois[0].get("value")
+            if is_doi(doi):
+                mapped_doi = {
+                    "identifier": doi,
+                }
+                if doi.startswith(DATACITE_PREFIX):
+                    mapped_doi["provider"] = "datacite"
+                else:
+                    mapped_doi["provider"] = "external"
+                return mapped_doi
+            else:
+                self.metadata_errors.append(
+                    f"DOI validation failed. Value: {doi}. INSPIRE record #{self.inspire_metadata.get('control_number')}."
+                )
+                return None
+
     def _transform_alternate_identifiers(self):
         """Mapping of alternate identifiers."""
         identifiers = []
         inspire_id = self.inspire_metadata.get("control_number")
+        RDM_RECORDS_IDENTIFIERS_SCHEMES = current_app.config[
+            "RDM_RECORDS_IDENTIFIERS_SCHEMES"
+        ]
+        IDENTIFIERS_SCHEMES_TO_DROP = ["SPIRES", "HAL", "OSTI", "SLAC", "PROQUEST"]
+
         try:
+            # persistent_identifiers
+            persistent_ids = self.inspire_metadata.get("persistent_identifiers", [])
+            for persistent_id in persistent_ids:
+                schema = persistent_id.get("schema").lower()
+                value = persistent_id.get("value")
+                if schema not in RDM_RECORDS_IDENTIFIERS_SCHEMES.keys():
+                    self.metadata_errors.append(
+                        f"Unexpected schema found in persistent_identifiers. Schema: {schema}, value: {value}. INSPIRE record id: {inspire_id}."
+                    )
+                else:
+                    identifiers.append({"identifier": value, "scheme": schema})
+
             # add INSPIRE id
             identifiers.append({"identifier": str(inspire_id), "scheme": "inspire"})
+
+            # external_system_identifiers
+            external_sys_ids = self.inspire_metadata.get(
+                "external_system_identifiers", []
+            )
+            for external_sys_id in external_sys_ids:
+                schema = external_sys_id.get("schema").lower()
+                if schema == "cds":
+                    schema = "lcds"
+                value = external_sys_id.get("value")
+                if schema.upper() in IDENTIFIERS_SCHEMES_TO_DROP:
```

Comment:
this condition should also apply to persistent_identifiers field, in case there is one id with a scheme which we don't want to keep